wpimath/src/main/java/edu/wpi/first/math/geometry/BoundVector2d.java

```diff
@@ -0,0 +1,191 @@
+package edu.wpi.first.math.geometry;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import edu.wpi.first.math.geometry.struct.BoundVector2dStruct;
+import edu.wpi.first.util.struct.StructSerializable;
+import java.util.Objects;
+
+/** Represents a bound vector in a vector field. Has an origin, and x/y component vectors. */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonAutoDetect(getterVisibility = JsonAutoDetect.Visibility.NONE)
+public class BoundVector2d implements StructSerializable {
+  private final Translation2d m_position;
+  private final Translation2d m_components;
+
+  /**
+   * Constructs a vector with the specified position and component vectors.
+   *
+   * @param position The position of the vector.
+   * @param components The X/Y component vectors.
+   */
+  public BoundVector2d(
+      @JsonProperty(required = true, value = "position") Translation2d position,
+      @JsonProperty(required = true, value = "components") Translation2d components) {
+    this.m_position = position;
+    this.m_components = components;
+  }
+
+  /**
+   * Constructs a bound vector with the specified position, direction, and magnitude.
+   *
+   * @param position The position of the vector.
+   * @param direction The direction of the vector.
+   * @param magnitude The magnitude of the vector.
+   */
+  public BoundVector2d(Translation2d position, Rotation2d direction, double magnitude) {
+    this(position, new Translation2d(magnitude, direction));
+  }
+
+  /**
+   * Constructs a vector from a start and end point.
+   *
+   * @param start The start point.
+   * @param end The end point.
+   * @return A new Vector2d with the correct position and components.
+   */
+  public static BoundVector2d fromStartEndPoints(Translation2d start, Translation2d end) {
+    return new BoundVector2d(start, end.minus(start));
+  }
+
+  /**
+   * Returns the position of the vector.
+   *
+   * @return The position of the vector.
+   */
+  @JsonProperty
+  public Translation2d getPosition() {
+    return m_position;
+  }
+
+  /**
+   * Returns the X component of the vector's position
+   *
+   * @return The Y component of the vector's position
+   */
+  public double getXPos() {
+    return m_position.getX();
+  }
+
+  /**
+   * Returns the Y component of the vector's position
+   *
+   * @return The Y component of the vector's position
+   */
+  public double getYPos() {
+    return m_position.getY();
+  }
+
+  /**
+   * Returns the components of the vector.
+   *
+   * @return The components of the vector.
+   */
+  @JsonProperty
+  public Translation2d getComponents() {
+    return m_components;
+  }
+
+  /**
+   * Returns the X component of the vector.
+   *
+   * @return The X component of the vector.
+   */
+  public double getXComponent() {
```

Comment:
This isn't really saving much typing over `getComponents().getX()`. At least `Pose2d.getX()` is a lot shorter than `Pose2d.getTranslation().getX()`.

Reply:
I personally think we should still include this, it saves a little bit of typing even if it isn't much.